src/EditorFeatures/CSharpTest/SymbolKey/SymbolKeyTests.cs

```diff
@@ -63,5 +66,147 @@ public sealed class ConditionalWeakTableTest<TKey, TValue> /*: IEnumerable<KeyVa
 
             Assert.Equal(method, resolved);
         }
+
+        [Fact]
+        [WorkItem(1178861, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1178861")]
+        [WorkItem(1192188, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1192188")]
+        [WorkItem(1192486, "https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1192486")]
+        public async Task ResolveBodySymbolsInMultiProjectReferencesToOriginalProjectAsync()
+        {
+            var random = new Random(Seed: 0);
+
+            // try to trigger race caused by ability to find reference in multiple potential projects depending on what
+            // order things are in in internal collections.  This test was always able to hit the issue prior to the fix
+            // going in, but does not hit it with the fix.  While this doesn't prove the race is gone, it strongly
+            // implies it.
+            for (var i = 0; i < 100; i++)
+            {
+                using var workspace = GetWorkspace();
+                var solution = workspace.CurrentSolution;
+
+                var bodyProject = solution.Projects.Single(p => p.AssemblyName == "BodyProject");
+                var referenceProject = solution.Projects.Single(p => p.AssemblyName == "ReferenceProject");
+
+                var (bodyCompilation, referenceCompilation) = await GetCompilationsAsync(bodyProject, referenceProject);
+                var (bodyLocalSymbol, referenceAssemblySymbol) = await GetSymbolsAsync(bodyCompilation, referenceCompilation);
+
+                var (bodyLocalProjectId, referenceAssemblyProjectId) = GetOriginatingProjectIds(solution, bodyLocalSymbol, referenceAssemblySymbol);
+
+                Assert.True(bodyProject.Id == bodyLocalProjectId, $"Expected {bodyProject.Id} == {bodyLocalProjectId}. {i}");
+                Assert.Equal(referenceProject.Id, referenceAssemblyProjectId);
+            }
+
+            return;
+
+            TestWorkspace GetWorkspace()
+            {
+                var bodyProject = @"
+    <Project Language=""C#"" AssemblyName=""BodyProject"" CommonReferences=""true"">
+        <Document>
+class Program
+{
+    void M()
+    {
+        int local;
+    }
+}
+        </Document>
+    </Project>";
+                var referenceProject = @"
+    <Project Language=""C#"" AssemblyName=""ReferenceProject"" CommonReferences=""true"">
+        <ProjectReference>BodyProject</ProjectReference>
+        <Document>
+        </Document>
+    </Project>";
+
+                // Randomize the order of the projects in the workspace.
+                if (random.Next() % 2 == 0)
```

Comment:
i needed the iteration count. i don't always hit this on the same iteration.  i think it depends on where a GC might happen, and thus what path/data-structure we might find it in.\

i could switch to a theory... but i guess i don't really get how that changes anything.  either we fail here, or we don't.  if we fail we have a race/issue we need to resolve.  since it's non deterministic, we'd still need to just isolate manually and through invesitigation.